### DISCUSSION TITLE
Which fields are searched by "quick search"

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14573] [Help Wanted!] [answered]
It looks like the "quick search" box at the top of the devices page only searches device names.  Is that correct behavior and is there some way to configure it?  I was hoping to search all visible fields including description, manufacturer, location, etc.
I'm running v3.6.7 if that matters.

**Comment by candlerb** [comment]
See https://github.com/netbox-community/netbox/blob/v3.6.7/netbox/dcim/search.py#L37-L46 to see which fields on Device are indexed (this is internal to Netbox and it's not supported to change this, it's just for reference).

* Device Description should be indexed. If you can demonstrate that it is not, then please raise an issue with steps to reproduce.
* Manufacturer and Location: these are different objects related to Device. It would drown out the search results if, for example, a search on "Cisco" were to include all devices made by Cisco.

However, the manufacturer Cisco should come up in the search results. If you click on it you'll go to `/dcim/manufacturers/NNN/`. This will show you the number of models (device types) for that manufacturer:

![image](https://github.com/netbox-community/netbox/assets/44789/c5fd8090-459e-4bbe-b512-32e173392cbf)

Click on the number and you'll get a list of those device types, with in turn a count of the number of instances of each, which you can click on to get the list of devices.

**Comment by llamafilm** [comment]
@candlerb I think you're talking about the search box at the top of the page.  I'm talking about the box labeled "quick search".  When I'm looking for a device, my instinct is to start at the Sites page, then click the Location (aka room), and then I see the list of devices in that room, I want to quickly find something in there.  In this context, I don't think full-text search would be overwhelming, since it's limited to devices only, with the "Location" filter already applied.

![image](https://github.com/netbox-community/netbox/assets/3991046/981ea9b3-4c1b-4fee-a5e9-665d0a069838)

**Comment by candlerb** [comment, marked as answer]
Ah, sorry. With that one, the page contains:

```
<input type="search" results="5" name="q" id="quicksearch" class="form-control" placeholder="Quick search" hx-get="" hx-target="#object_list" hx-trigger="keyup changed delay:500ms, search">
```

As far as I can tell from browser dev tools, it makes a query to `/dcim/devices/?q=XXX`, so you can look in the filtersets to see what that matches.

**Reply by candlerb** [reply]
Here it is: https://github.com/netbox-community/netbox/blob/v3.6.7/netbox/dcim/filtersets.py#L1013-L1024

```
        return queryset.filter(
            Q(name__icontains=value) |
            Q(serial__icontains=value.strip()) |
            Q(inventoryitems__serial__icontains=value.strip()) |
            Q(asset_tag__icontains=value.strip()) |
            Q(comments__icontains=value) |
            Q(primary_ip4__address__startswith=value) |
            Q(primary_ip6__address__startswith=value)
        ).distinct()
```

I agree that for consistency, "description" ought to be searched here. You could raise a feature request for that.

However, apart from `inventoryitems__serial` and the IP addresses, you can see that everything is a field local to the devices table.

The more extra things that are searched for in other tables, the slower it will become. If you add `device_type__model__icontains`, so that typing "NCS" will include all devices with "NCS" in their model name, that will require a table join. If you add `device_type__manufacturer__name__icontains` (so that you can type "Cisco") then that's two table joins.  `platform__name__icontains` (so that "XR" finds IOS-XR devices) would be another. And so on.

In general, I'd say searches like this are better done by using the "Filters" tab. In any case, since it's explicit which field you're searching within (model, manufacturer, platform etc) you're more likely to get a better set of matches.

I suppose it would be possible to implement a github-style query language, so that you could type `platform:xr` for example. But I'm not sure that would get used very much. By the time you've typed `manufacturer:cisco` it would have been quicker to go to Filters.

**Comment by llamafilm** [comment]
Ok, yeah that makes sense, I wouldn't want to slow it down.  I've opened #14596 to just add description.
